### DISCUSSION TITLE
Upgraded to  2.3.70  Receiving the following error

### DISCUSSION THREAD

**Discussion by lerou114** [Security-Onion-Solutions/securityonion#5267] [General] [answered]
I am running a Security onion deployed as a distributed environment with a standalone fleet server and when running SOUP  upgrading 2.3.61 to 2.370 I receive this error:

[ERROR   ] Rendering exception occurred
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 497, in render_jinja_tmpl
    output = template.render(**decoded_context)
  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/usr/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 106, in top-level template code
  File "/usr/lib/python3.6/site-packages/jinja2/sandbox.py", line 407, in getattr
    value = getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'fleet'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 261, in render_tmpl
    output = render_str(tmplstr, context, tmplpath)
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 504, in render_jinja_tmpl
    raise SaltRenderError("Jinja variable {}{}".format(exc, out), buf=tmplstr)
salt.exceptions.SaltRenderError: Jinja variable 'dict object' has no attribute 'fleet'
[CRITICAL] Rendering SLS 'base:grafana' failed: Jinja variable 'dict object' has no attribute 'fleet'
local:
    Data failed to compile:
----------
    Rendering SLS 'base:grafana' failed: Jinja variable 'dict object' has no attribute 'fleet'

**Comment by m0duspwnens** [comment]
@lerou114 Thanks for the bug report. We will look into this issue.

**Reply by m0duspwnens** [reply]
@lerou114 We currently have a hotfix in the pipeline, but if you don't have an airgapped install, and would like to test the fix prior to release, that could be done with `BRANCH=hotfix/2.3.70 soup`. This has been tested while developing it, and functioned as expected, but has not run through our full test suite yet, so please use caution if deciding to test the hotfix early.

The issue was being caused by the Grafana state attempting to create a Fleet dashboard since the state creates a dashboard for each node type in the environment. Since we haven't defined the Fleet dashboard yet, the state failed attempting to lookup the parameters for that dashboard. The solution was to only allow the Grafana state to create a dashboard for a node type if that node type has parameters defined for it.

Thanks again for the bug report.

**Comment by dougburks** [comment, marked as answer]
We've released this hotfix:
https://blog.securityonion.net/2021/08/security-onion-2370-grafana-hotfix-now.html
